.github/workflows/release.yaml

```diff
@@ -1,48 +1,94 @@
 name: release
 on:
   push:
+    branches:
+      - 'master'
     tags:
       - 'v[0-9]+.[0-9]+.[0-9]+'
+  pull_request: {}
 defaults:
   run:
     shell: bash
 jobs:
-  create:
+  release:
     runs-on: ubuntu-latest
-    outputs:
-      upload_url: ${{ steps.release.outputs.upload_url }}
-    steps:
-      - uses: actions/create-release@v1
-        id: release
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
-        with:
-          draft: true
-          tag_name: ${{ github.ref }}
-          release_name: ${{ github.ref }}
-  assets:
-    needs: create
-    strategy:
-      matrix:
-        os:
-          - ubuntu-latest
-          - macos-latest
-          - windows-latest
-    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Any value in adding a comment here as to why this is necessary? It looks like the [goreleaser workflow example](https://github.com/goreleaser/goreleaser-action#workflow) calls out this is necessary to have changelogs work correctly, but it wasn't immediately obvious to me and I had to go poke around that linked README.md.